storage/innobase/handler/ha_innodb.cc

```diff
@@ -117,6 +120,7 @@ this program; if not, write to the Free Software Foundation, Inc.,
 #include "row0ext.h"
 
 #include <limits>
+#include <myisamchk.h>                          // TT_FOR_UPGRADE
```

Comment:
This was contributed by @montywi. I think we may want to avoid a larger refactoring in maintenance releases. It is not nice, but I can live with that.

Reply:
Agree. Should be fine if we do it some day :) hopefully.